packages/runtime/src/helpers/edge.ts

```diff
@@ -276,12 +300,24 @@ export const writeEdgeFunctions = async (netlifyConfig: NetlifyConfig) => {
     // Older versions of the manifest format don't have the functions field
     // No, the version field was not incremented
     if (typeof middlewareManifest.functions === 'object') {
+      // When using the app dir, we also need to check if the EF matches a page
+      const appPathRoutesManifest = await loadAppPathRoutesManifest(netlifyConfig)
```

Comment:
Does this mean that the `app` dir functionality is only available if using [Next.js's Middleware](https://nextjs.org/docs/advanced-features/middleware)?

Reply:
No, those are in `middlewareManifest.middleware`. The `middlewareManifest.functions` object holds Next.js Edge Functions, which are equivalent to our post-cache edge functions. They're generated automatically when using appDir.